stl/inc/chrono

```diff
@@ -609,8 +610,22 @@ namespace chrono {
         static constexpr bool is_steady = true;
 
         _NODISCARD static time_point now() noexcept { // get current time
-            const long long _Freq = _Query_perf_frequency(); // doesn't change after system boot
-            const long long _Ctr  = _Query_perf_counter();
+#if (defined(_M_ARM) || defined(_M_IX86) || defined(_M_ARM64) || defined(_M_X64)) && !defined(_M_CEE_PURE)
+            // ad-hoc <atomic> load/compare_exchange_strong with std::memory_order_relaxed
+            // avoid dependency on real <atomic>
+            static volatile long long _Freq_cached = -1; // doesn't change after system boot
+            // _Freq_cached must always be accessed via an Interlocked function,
+            // But it is accessed via __iso_volatile_load64, which is also proper way
+#pragma warning(suppress : 28112)
+            long long _Freq = __iso_volatile_load64(&_Freq_cached);
+            if (_Freq == -1) {
+                _Freq = _Query_perf_frequency();

```

Comment:
Should the caching be done in _Query_perf_frequency() / xtime.cpp itself so that existing code gets the perf benefit too?

Reply:
Great, this way I can just take real <atomic>